### DISCUSSION TITLE
Make AeroTab it's own repository

### DISCUSSION THREAD

**Discussion by Ovewh** [NorESMhub/CAM#104] [Ideas]
AeroTab, which is the code that generates the optical tables used in Oslo_Aero,  currently lives in the cam/tools directory (not in the noresm_develop branch). 
Because lacking of documentation and being hard to use, the tables we are using in Oslo_Aero have not really been look at for the past six years.

The code needs to be refactored to become more configurable and better documented, a end goal could be to make a python wrapper and create something like a pyAeroTab. In my opinion this could make AeroTab into a great tool exploring impact of e.g. aerosol processing (aging and hygroscopic growth) on size distribution and a useful tool for e.g. teaching purposes. 

TL;DR Initiate a process of refactoring AeroTab and make into its own repository.

**Comment by gold2718** [comment]
I second this and can also do the set up if desired.
In the end, it folks want to be available with CAM, it can be "brought back" as an external.

Note, making AeroTab its own repository lowers the cost of development since it is much easier to up date the code (e.g., process a pull request). It is easier to review a smaller code base. It would also make it easier to add some tests and checks to make sure everything is always working properly.

**Reply by Ovewh** [reply]
Yes it would be great if you can set up the workflow.

**Reply by gold2718** [reply]
Just to check, do you agree with these statements?

- The new repo will be called `AeroTab`
- The original source for `AeroTab` will come from `cam_cesm2_1_rel_05-Nor_v1.0.5` (NorESM2 release version)
- The `AeroTab` source will be removed from the NorESM2.1 release branch
- `AeroTab` repo will start with a single branch, `main`, containing the original source

Let me know if I have any of these wrong.

**Reply by MichaelSchulzMETNO** [reply]
@oyvindseland - would you agree also?

**Comment by MichaelSchulzMETNO** [comment]
That sounds good to have it as a stand alone, but within-NorESMhub github directory.

**Comment by oyvindseland** [comment]
yes

fre. 20. okt. 2023 kl. 16:39 skrev Michael Schulz ***@***.***

**Reply by oyvindseland** [reply]
Sorry I missed this question. To elaborate. Yes it is good to have it in separate repository in particular since it is entirely independent of CAM / NorESM. There are some sharing of code with Oslo-Aero but embedded in different subroutines.

**Reply by Ovewh** [reply]
Do you mean code duplication with Oslo_Aero? Or actually sharing subroutines such that it would not compile if it does not know about Oslo_Aero?

**Reply by gold2718** [reply]
It looks like the dependent code has all been copied into the tools/AeroTab folder. 
I think this is mainly commondefinitions.F90 (which has been updated and documented in Oslo_Aero.
I also see a copy of shr_kind_mod.F90.

The code builds fine just using the files in tools/AeroTab.

I think I can go ahead and do the separation as part of the NorESM2.1 release.

**Reply by oyvindseland** [reply]
No AeroTab does not depend on subroutines in Oslo-Aero so it is code duplication and not code sharing. As long as AeroTab is used for making tables in NorESM one should remember to update the code both places. One example is assumed accommodation coefficient for sulfate and organic vapor deposition on to mineral dust particles.

**Comment by gold2718** [comment]
This repo now [exists](https://github.com/NorESMhub/AeroTab), it will probably be included as an external beginning with NorESM2.3 (which will use CAM externals, unlike NorESM2.1).

**Comment by oyvindseland** [comment]
AeroTab has now its own repository

https://github.com/NorESMhub/AeroTab
